apiary-metastore-listener/src/main/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListener.java

```diff
@@ -0,0 +1,144 @@
+
+  private static final Logger log = LoggerFactory.getLogger(ApiarySnsListener.class);
+
+  private static final String TOPIC_ARN = System.getenv("SNS_ARN");
+  private AmazonSNSClient snsClient;
+
+  public ApiarySnsListener(Configuration config) {
+    super(config);
+    // create a new SNS client and set endpoint
+    snsClient = new AmazonSNSClient();
+    snsClient.setRegion(RegionUtils.getRegion(System.getenv("AWS_REGION")));
+    log.debug("ApiarySnsListener created ");
+  }
+
+  ApiarySnsListener(Configuration config, AmazonSNSClient snsClient) {
+    super(config);
+    this.snsClient = snsClient;
+    log.debug("ApiarySnsListener created ");
+  }
+
+  @Override
+  public void onCreateTable(CreateTableEvent event) throws MetaException {
+    if (event.getStatus() == false) {
+      return;
+    }
+    publishEvent("CREATE_TABLE", event.getTable(), null, null, null);
+  }
+
+  @Override
+  public void onDropTable(DropTableEvent event) throws MetaException {
+    if (event.getStatus() == false) {
+      return;
+    }
+    publishEvent("DROP_TABLE", event.getTable(), null, null, null);
+  }
+
+  @Override
+  public void onAlterTable(AlterTableEvent event) throws MetaException {
+    if (event.getStatus() == false) {
+      return;
+    }
+    publishEvent("ALTER_TABLE", event.getNewTable(), event.getOldTable(), null, null);
+  }
+
+  @Override
+  public void onAddPartition(AddPartitionEvent event) throws MetaException {
+    if (event.getStatus() == false) {
+      return;
+    }
+    Iterator<Partition> partitions = event.getPartitionIterator();
+    while (partitions.hasNext()) {
+      publishEvent("ADD_PARTITION", event.getTable(), null, partitions.next(), null);
+    }
+  }
+
+  @Override
+  public void onDropPartition(DropPartitionEvent event) throws MetaException {
+    if (event.getStatus() == false) {
+      return;
+    }
+    Iterator<Partition> partitions = event.getPartitionIterator();
+    while (partitions.hasNext()) {
+      publishEvent("DROP_PARTITION", event.getTable(), null, partitions.next(), null);
+    }
+  }
+
+  @Override
+  public void onAlterPartition(AlterPartitionEvent event) throws MetaException {
+    if (event.getStatus() == false) {
+      return;
+    }
+    publishEvent("ALTER_PARTITION", event.getTable(), null, event.getNewPartition(), event.getOldPartition());
+  }
+
+  private void publishEvent(String event_type, Table table, Table oldtable, Partition partition, Partition oldpartition)
+    throws MetaException {
+
+    JSONObject json = new JSONObject();
+    json.put("eventType", event_type);
+    json.put("dbName", table.getDbName());
+    json.put("tableName", table.getTableName());
+    if (oldtable != null) {
+      // TODO = discuss - can we change the case on this to be consistent? i.e. "oldTableName"
+      json.put("oldtableName", oldtable.getTableName());
+    }
+    if (partition != null) {
+      // TODO = discuss - can we change the case on this to be consistent? i.e. "partition"
```

Comment:
@rpoluri are you ok with me changing this to "partition"?

Reply:
sounds good.